File: app/src/main/java/com/stn/carterminal/network/request/ChangeVehiclePosition.java
package com.stn.carterminal.network.request;

import lombok.Data;

@Data
public class ChangeVehiclePosition {
    private Long id;
    private final String vehiclePosition = "TERMINAL_IN";
    private String uhfTag;
    private String vehicleIdNumber;
    private String description;
    private Long vehicleClassId;
    private String vehicleManifestStatus;
    private boolean isDataVehicleChanged;

    public ChangeVehiclePosition(Long id, String uhfTag, String vehicleIdNumber, String description, Long vehicleClassId, String vehicleManifestStatus, boolean isDataVehicleChanged) {
        this.id = id;
        this.uhfTag = uhfTag;
        this.vehicleIdNumber = vehicleIdNumber;
        this.description = description;
        this.vehicleClassId = vehicleClassId;
        this.vehicleManifestStatus = vehicleManifestStatus;
        this.isDataVehicleChanged = setDataVehicleChanged(isDataVehicleChanged);
    }

    private boolean setDataVehicleChanged(boolean isDataVehicleChanged) {
        if (vehicleManifestStatus.equals("ADD")) {
            return false;
        } else {
            return isDataVehicleChanged;
        }
    }
}

File: app/src/main/java/com/stn/carterminal/network/request/ChangeVehiclePosition.kt
package com.stn.carterminal.network.request

class ChangeVehiclePosition(var id: Long, var uhfTag: String, var vehicleIdNumber: String, var description: String, var vehicleClassId: Long, var vehicleManifestStatus: String, var isDataVehicleChanged: Boolean) {
    val vehiclePosition: String = "TERMINAL_IN"

    init {
        isDataVehicleChanged = setDataVehicleChanged(isDataVehicleChanged)
    }

    private fun setDataVehicleChanged(isDataVehicleChanged: Boolean): Boolean {
        when (vehicleManifestStatus) {
            "ADD" -> return false
            else -> return isDataVehicleChanged
        }
    }
}